mlflow/pipelines/utils/step.py

```diff
@@ -76,9 +76,15 @@ def display_html(html_data: str = None, html_file_path: str = None) -> None:
         import shutil
         import subprocess
 
-        if os.path.exists(html_file_path) and shutil.which("open") is not None:
+        # Use xdg-open in Linux environment
+        if shutil.which("xdg-open") is not None:
+            open_tool = "xdg-open"
+        elif shutil.which("open") is not None:
+            open_tool = "open"
+
+        if os.path.exists(html_file_path) and open_tool is not None:
             _logger.info(f"Opening HTML file at: '{html_file_path}'")
-            subprocess.run(["open", html_file_path], check=True)
+            subprocess.run([open_tool, html_file_path], check=True)
```

Comment:
Should this line be wrapped with `try-catch` because `open` or `xdg-open` might fail if a web browser is unavailable?

Reply:
We should catch error and log warning if it failed.

Reply:
Good suggestion.  That probably would have resolved the original issue I ran into with `open` in the first place.  

I updated the block to log the exception as a warning without any further action.  Let me know if you think there is any additional action that should be taken in the exception.